packages/x-date-pickers/src/PickersLayout/PickersLayout.types.ts

```diff
@@ -119,10 +120,10 @@ export interface PickersLayoutProps<TValue, TDate, TView extends DateOrTimeViewW
   slotProps?: PickersLayoutSlotProps<TValue, TDate, TView>;
 }
 
-export interface SubComponents {
-  toolbar: React.ReactNode;
+export interface SubComponents<TValue> {
+  toolbar: React.ReactElement<ExportedBaseToolbarProps> | null;
   content: React.ReactNode;
-  tabs: React.ReactNode;
-  actionBar: React.ReactNode;
-  shortcuts: React.ReactNode;
+  tabs: React.ReactElement<ExportedBaseTabsProps> | null;
+  actionBar: React.ReactElement<PickersActionBarProps>;
```

Comment:
Needed to access `actionBar.props.actions`.
Changed the rest given the chance. 😃